src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -491,4 +492,41 @@ void setCrossExpansionProperties(final CrossExpansionProperties crossExpansionPr
 		this.crossExpansionProperties = crossExpansionProperties;
 	}
 
+	@Override
+	public List<AttributeDTO> getGermplasmAttributes() {
+		final Set<String> types = new HashSet<>();
+		return this.germplasmDataManager.getUserDefinedFieldByFieldTableNameAndType(UDTableType.ATRIBUTS_ATTRIBUTE.getTable(),
+			ImmutableSet.of(UDTableType.ATRIBUTS_ATTRIBUTE.getType(), UDTableType.ATRIBUTS_PASSPORT.getType()))
+			.stream()
+			.map(userDefinedField -> {
+				final AttributeDTO attributeDTO =
+					new AttributeDTO();
+				attributeDTO.setAttributeDbId(userDefinedField.getFldno());
+				attributeDTO.setAttributeName(userDefinedField.getFname());
+				attributeDTO.setAttributeCode(userDefinedField.getFcode());
+				return attributeDTO;
+			})
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<Method> getAllBreedingMethods() {
+		return this.germplasmDataManager.getAllMethods();
+	}
+
+	@Override
+	public List<GermplasmName> getGermplasmNames() {
```

Comment:
it maybe duplicated, read comment in the interface

Reply:
removed.